docs/src/developing/clients/jsonrpc-api.md

```diff
@@ -1583,7 +1583,10 @@ Returns the leader schedule for an epoch
 #### Parameters:
 
 - `<u64>` - (optional) Fetch the leader schedule for the epoch that corresponds to the provided slot. If unspecified, the leader schedule for the current epoch is fetched
-- `<object>` - (optional) [Commitment](jsonrpc-api.md#configuring-state-commitment)
+- `<object>` - (optional) Configuration object containing the following optional fields:
+  - (optional) [Commitment](jsonrpc-api.md#configuring-state-commitment)
+  - (optional) `limit: <number>` - Limit the number of upcoming slots to fetch from the queried leader schedule. (Max: 432,000)
```

Comment:
Hmm, I'm not liking now how `limit` causes the change in starting slot index from 0 to the current slot number.

What if we added something new instead, `getLeaderScheduleWithLimit`, with params `start slot` and `limit` (same params as [getConfirmedBlocksWithLimit](https://docs.solana.com/developing/clients/jsonrpc-api#getconfirmedblockswithlimit))

`limit` should less than 5,000 or so (so max ~250kb response) and just returns a vector of pubkeys for the slots (no dictionary), which makes it easy to return results across an epoch boundary as well

Reply:
That's fair. The client experience of getting pubkeys instead of slot indexes would be much better. I'll mark the pre-existing `getLeaderSchedule` method as deprecated while I'm at it

Reply:
For larger ranges, like what's needed to figure block production over an epoch,  the existing `getLeaderSchedule` is still nice because it avoids an RPC response of ~19MB.  So I think there's room for both to exist

Reply:
Cool, left it alone. I went with a new `GetSlotLeaders` method which fits better with your idea I think